retrievalmarket/impl/clientstates/client_states.go

```diff
@@ -89,42 +92,97 @@ func Ongoing(ctx fsm.Context, environment ClientDealEnvironment, deal rm.ClientD
 
 // ProcessPaymentRequested processes a request for payment from the provider
 func ProcessPaymentRequested(ctx fsm.Context, environment ClientDealEnvironment, deal rm.ClientDealState) error {
-	// see if we need to send payment
-	if deal.TotalReceived-deal.BytesPaidFor >= deal.CurrentInterval ||
-		deal.AllBlocksReceived ||
-		deal.UnsealPrice.GreaterThan(deal.UnsealFundsPaid) {
+	// If the unseal payment hasn't been made, we need to send funds
+	if deal.UnsealPrice.GreaterThan(deal.UnsealFundsPaid) {
+		log.Debugf("client: payment needed: unseal price %d > unseal paid %d",
+			deal.UnsealPrice, deal.UnsealFundsPaid)
+		return ctx.Trigger(rm.ClientEventSendFunds)
+	}
+
+	// If all bytes received have been paid for, we don't need to send funds
+	if deal.BytesPaidFor >= deal.TotalReceived {
+		log.Debugf("client: no payment needed: bytes paid for %d >= bytes received %d",
+			deal.BytesPaidFor, deal.TotalReceived)
+		return nil
```

Comment:
Why not trigger a `ClientEventPaymentNotSent ` here to move out to the `Ongoing ` state ?